addons/delivery/models/stock_picking.py

```diff
@@ -44,15 +45,16 @@ def _compute_weight_uom_name(self):
 class StockPicking(models.Model):
     _inherit = 'stock.picking'
 
-
     @api.depends('move_line_ids', 'move_line_ids.result_package_id')
     def _compute_packages(self):
-        for package in self:
-            packs = set()
-            for move_line in package.move_line_ids:
-                if move_line.result_package_id:
-                    packs.add(move_line.result_package_id.id)
-            package.package_ids = list(packs)
+        picking_map = defaultdict(list)
+        data = self.env['stock.move.line'].read_group(
+            [('picking_id', 'in', self.ids), ('result_package_id','!=',False)], 
+            ['picking_id', 'result_package_id'], ['picking_id', 'result_package_id'], lazy=False)
```

Comment:
you can make the group directly in sql with array_agg
```suggestion
            ['picking_id', 'result_package_id'], ['picking_id', 'result_package_id:array_agg'], lazy=False)
```
Also could you give some figures about the time saved by this patch in the PR description ? Thanks a lot